usb: sam0: Add support for SAME54

benpicco opened the pull request:
The USB peripheral is the same as in previous SAM0 MCUs, but the clock setup is different.

split off from #14685

jfischer-phytec-iot commented on drivers/usb/device/usb_dc_sam0.c:
```diff
@@ -132,6 +132,9 @@ static void usb_sam0_load_padcal(void)
 	u32_t pad_trim;
 
+#ifdef USB_FUSES_TRANSN_ADDR
+	pad_transn = *(uint32_t *)USB_FUSES_TRANSN_ADDR;
```
This change is not explained in the commit message. Where is USB_FUSES_TRANSN_ADDR defined, HAL?

benpicco replied:
Yes, it's in the [`component/nvmctrl.h`](https://github.com/zephyrproject-rtos/hal_atmel/blob/master/asf/sam0/include/same54/component/nvmctrl.h#L743) header file.

jfischer-phytec-iot commented on drivers/usb/device/usb_dc_sam0.c:
```diff
@@ -203,10 +235,18 @@ int usb_dc_attach(void)
-		    DT_INST_0_ATMEL_SAM0_USB_IRQ_0_PRIORITY,
-		    usb_sam0_isr, 0, 0);
-	irq_enable(DT_INST_0_ATMEL_SAM0_USB_IRQ_0);
+#ifdef DT_INST_0_ATMEL_SAM0_USB_IRQ_0
```
This change is also not explained in the commit message.

jfischer-no reviewed (changes requested):
Please split it in three commits: changes for PAD? settings, changes for clock setup, and changes for IRQ connect and enable.

benpicco committed:
usb: sam0: Support more then one USB IRQ

Newer SoCs like SAME54 have multiple IRQs for the USB peripheral.
Automatically enable all IRQs configured in the DTS.

Signed-off-by: Benjamin Valentin <benpicco@googlemail.com>

benpicco committed:
usb: sam0: Add support for MCLK

On newer sam0 SoCs peripherals are enabled through the MCLK
instead of the PM register.

Use the MCLK register if it's availiable.

Signed-off-by: Benjamin Valentin <benpicco@googlemail.com>

benpicco committed:
usb: sam0: use TRIM registers when available

Newer SoCs like SAME54 provide dedicated registers to store
the USB trim configuration.

Use those registers if they are defined in ASF to access the
calibration data.

Signed-off-by: Benjamin Valentin <benpicco@googlemail.com>

benpicco commented:
Ok, I've split up the commit.
